MarkdownView

marcopeg opened the pull request:
added a MarkdownView class to implement full view markdown rendering.

thanks to @dereuromark for suggestion!

marcopeg committed:
added MarkdownView

dereuromark commented:
I guess a test case for it would be nice :)

marcopeg commented:
@dereuromark you're right.

I have a personal task to learn to write CakeTests both for this PR and for my CustomValidation CakePHP's one!

I'm from Italy, this weekend will rain all the time. I think I will find the time to write some tests!

bye!

dereuromark commented:
tip: the core tests for View classes are great to learn how to start.

dereuromark commented on View/MarkdownView.php:
```diff
@@ -0,0 +1,118 @@
+ * @blogPost:	http://movableapp.com/2012/11/cakephp-markdown-plugin/
+ */
+
+App::import( 'Vendor', 'Markdown.markdown' );
```
why are you adding extra spaces? should be `App::import('Vendor', 'Markdown.markdown');`. Same for the rest of the file :)
There are also a lot of other violations of cake coding standards.

marcopeg replied:
Ho mark,
I really can't understand CakePHP code stantards... I like so much a better spaced code with comments and other useful code spaces (like alignments).

When writing PR to the CakePHP core I will follow CakePHP code standards... but for outside code I prefer to improve readability.

dereuromark commented:
Well, the thing is, you might be outside the core, but you are PRing for someone elses code. You will have to check with Hyra then if he finds that acceptable. Most wouldn't I imagine.
Imagine everyone using his own standards for PR to other peoples work. Chaos.

marcopeg commented:
you're right, I understand.
for this PR i know hyra accepts my code. he accepted it before this PR.

But honestly, do you find any difficult reading my code?

I find difficult to read CakePHP core's sources... no spaces, no comments, long long conditional braces without any explanations...

why?

(but I think we're going OT of this PR)

marcopeg committed:
added events

two events into the parser method allow external components to enhance
variables syntax.

marcopeg committed:
fix String::flatten warning...

marcopeg committed:
just some variations

marcopeg committed:
normalizations

now both helper and view renderers implement "parseViewVars()" method
from a static library MarkdownUtil.

this means you can use special placeholders (implemented by external
events!) in everywhere.

Hyra commented:
Hey @thepeg 

Thanks for the addition. I will see if i have time this weekend to test it out.
@dereuromark has a point about the space-convention thing, which i have been ignoring lately (my bad)

Thanks again! To be continued.

marcopeg commented:
I miss standards too. my lack.
I'm learning about test cases to commit them to the repo!

I home @dereuromark will help us to complete this plugin with test cases!

marcopeg committed:
reupdate

Commit:
fix
